pkg/operation/botanist/component/kubescheduler/kube_scheduler.go

```diff
@@ -122,15 +125,17 @@ func New(
 	image string,
 	replicas int32,
 	config *gardencorev1beta1.KubeSchedulerConfig,
+	runtimeKubernetesVersion *semver.Version,
 ) Interface {
 	return &kubeScheduler{
-		client:         client,
-		namespace:      namespace,
-		secretsManager: secretsManager,
-		version:        version,
-		image:          image,
-		replicas:       replicas,
-		config:         config,
+		client:                        client,
+		namespace:                     namespace,
+		secretsManager:                secretsManager,
+		version:                       version,
+		image:                         image,
+		replicas:                      replicas,
+		config:                        config,
+		runtimeVersionGreaterEqual123: versionutils.ConstraintK8sGreaterEqual123.Check(runtimeKubernetesVersion),
```

Comment:
As this version is not normalized, we'll probably run again into issues mentioned [here](https://github.com/gardener/gardener/pull/6468), i.e. `versionutils.ConstraintK8sGreaterEqual123.Check("v1.23.0-xyz")` will be `false`. It's probably the same for the checks in https://github.com/gardener/gardener/pull/6982.

See also https://go.dev/play/p/6xkL0cepY8e

Reply:
Good catch, thanks!